lib/queue.js

```diff
@@ -423,6 +430,7 @@ Queue.prototype.run = function(concurrency){
   var promises = [];
   var _this = this;
 
+  // In case of connection loss, running `processStalledJobs` will repick jobs here.
   return this.processStalledJobs().then(function(){
```

Comment:
If multiple workers are starting up at various times, all with processStalledJobs turned off, running this means those stuck jobs will still get processed.